### PR TITLE
Support client-driven refresh of expiring credentials

### PR DESCRIPTION
**Pull request by michaelklishin**
## Proposed Changes

This introduces support for a new AMQP 0-9-1 extension, `connection.update-secret`, which is used to update authZ backend state when it can expire. Specifically we want to support client-driven JWT/OAuth 2 access token "renewal" without reconnection.

See https://github.com/rabbitmq/rabbitmq-auth-backend-oauth2/issues/28 for background.

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes issue #NNNN)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause an observable behavior change in existing systems)
- [ ] Documentation improvements (corrections, new content, etc)
- [ ] Cosmetic change (whitespace, formatting, etc)

## Checklist

- [x] I have read the `CONTRIBUTING.md` document
- [x] I have signed the CA (see https://cla.pivotal.io/sign/rabbitmq)
- [x] All tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works (in `rabbitmq-auth-backend-oauth2`)
- [ ] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in related repositories

## Further Comments

Part of https://github.com/rabbitmq/rabbitmq-auth-backend-oauth2/issues/28.

### CODE REVIEW

File: src/rabbit_access_control.erl [hunk 1]
```diff
@@ -217,3 +219,31 @@ check_access(Fun, Module, ErrStr, ErrArgs, ErrName) ->
...
     end.
+
+-spec update_state(User :: rabbit_types:user(), NewState :: term()) ->
+    {'ok', rabbit_types:auth_user()} |
+    {'refused', string(), [any()]} |
+    {'error', any()}.
+
+update_state(User = #user{authz_backends = Backends0}, NewState) ->
+    %% N.B.: we use foldl/3 and prepending, so the final list of
+    %% backends
```
Comment (acogoluegnes) [inline src/rabbit_access_control.erl]: Is the sentence complete?

File: test/unit_inbroker_parallel_SUITE.erl [hunk 2]
```diff
@@ -61,6 +61,9 @@ groups() ->
               login_of_passwordless_user,
               set_tags_for_passwordless_user
             ]},
+          {access_control, [parallel], [
+              state_update
```
Comment (acogoluegnes) [inline test/unit_inbroker_parallel_SUITE.erl]: We already discussed not to add this test, right?

Comment (michaelklishin) [reply test/unit_inbroker_parallel_SUITE.erl]: Yes, will remove it.

File: src/rabbit_access_control.erl (hunk 1 above)
Comment (michaelklishin) [reply src/rabbit_access_control.erl]: Corrected.

**Comment by michaelklishin** [comment]
We can drop permission cache if at least one backend supports credential expiration.

**Pull request merged**